sql/src/main/java/org/apache/druid/sql/calcite/aggregation/DimensionExpression.java

```diff
@@ -38,11 +39,27 @@ public DimensionExpression(
       final ValueType outputType
   )
   {
+    this(outputName, outputName, expression, outputType);
+  }
+
+  public DimensionExpression(
```

Comment:
It's not immediately clear when each constructor should be used. They need javadocs, or to be made private and replaced with nicely-named static factories, or (IMO ideally) both.